code/cutscene/cutscenes.cpp

```diff
@@ -48,38 +49,101 @@ void cutscene_close()
 		}
 }
 
+static cutscene_info *get_cutscene_pointer(char *cutscene_filename)
+{
+	for (int i = 0; i < Num_cutscenes; i++) {
+		if (!stricmp(cutscene_filename, Cutscenes[i].filename)) {
+			return &Cutscenes[i];
+		}
+	}
+
+	// Didn't find anything.
+	return NULL;
+}
+
+static void cutscene_info_init(cutscene_info *csni)
+{
+	csni->filename[0] = '\0';
+	csni->name[0] = '\0';
+	csni->description = NULL;
+	csni->flags = {};
+}
+
 // initialization stuff for cutscenes
-void cutscene_init()
+void parse_cutscene_table(const char* filename)
 {
-	atexit(cutscene_close);
+
 	char buf[MULTITEXT_LENGTH];
-	cutscene_info cutinfo;
+	cutscene_info csnt;
+	cutscene_info_init(&csnt);
+
+	cutscene_info *csnp;
+	bool create_if_not_found = true;
 
 	try
 	{
-		read_file_text("cutscenes.tbl", CF_TYPE_TABLES);
+		
+		read_file_text(filename, CF_TYPE_TABLES);
+
 		reset_parse();
 
 		// parse in all the cutscenes
-		Cutscenes.clear();
 		skip_to_string("#Cutscenes");
 		ignore_white_space();
 
-		bool isFirstCutscene = true;
-
 		while (required_string_either("#End", "$Filename:"))
 		{
 			required_string("$Filename:");
-			stuff_string(cutinfo.filename, F_PATHNAME, MAX_FILENAME_LEN);
+			stuff_string(csnt.filename, F_PATHNAME, MAX_FILENAME_LEN);
 
-			required_string("$Name:");
-			stuff_string(cutinfo.name, F_NAME, NAME_LENGTH);
+			if (optional_string("+nocreate")) {
+				if (!Parsing_modular_table) {
+					Warning(LOCATION, "+nocreate flag used for cutscene in non-modular table\n");
+				}
+				create_if_not_found = false;
+			}
+
+			// Does this cutscene exist already?
+			// If so, load this new info into it
+			csnp = get_cutscene_pointer(csnt.filename);
+			if (csnp != NULL) {
+				if (Parsing_modular_table) {
+					nprintf(
+						("Warning", "More than one version of cutscene %s exists; using newer version.\n", csnt.filename));
```

Comment:
Should this really be a warning?  In most cases a modular table modifying a previously parsed entry is a normal thing.

Reply:
Engine Wash parsing (where most of this was read and adapted from) has a similar warning which is why it also exists here. I don't care either way. Just trying to match the style elsewhere in the engine.

Reply:
Hmm.  I'd say engine washes are a special case, and the warning isn't needed here.  Looking through the code, engine washes are the only other place this sort of warning is displayed.